common/djangoapps/student/models.py

```diff
@@ -931,6 +931,9 @@ class CourseEnrollment(models.Model):
     # in the course (is_enrolled() will return False)
     is_active = models.BooleanField(default=True)
 
+    # If the user enrolled in this course through the Recommendation Engine recommendations
+    through_recommendations = models.BooleanField(default=False, null=False, blank=False)
```

Comment:
I really do not understand why do you need this field, could you please explain its logic? Second of all, say you do need it, it must be added to another table not this one, and thats because we need to separate concerns.